src/writer/deploy.py

```diff
@@ -134,53 +185,53 @@ def dictFromEnv(env: List[str]) -> dict:
-        start_time = datetime.now(pytz.timezone('UTC'))
-        build_time = start_time
-        with requests.post(
-            url = WRITER_DEPLOY_URL, 
-            headers = {
-                "Authorization": f"Bearer {token}",
-            },
-            files=files,
-            data={"envs": json.dumps(dictFromEnv(env))}
-        ) as resp:
-            try:
-                resp.raise_for_status()
-            except Exception as e:
-                print(resp.json())
-                raise e
-            data = resp.json()
-            build_id = data["buildId"]
-
-        print("Package uploaded. Building...")
-        status = "WAITING"
-        url = ""
-        while status not in ["COMPLETED", "FAILED"] and datetime.now(pytz.timezone('UTC')) < build_time + timedelta(minutes=5):
-            end_time = datetime.now(pytz.timezone('UTC'))
-            status, url = check_service_status(token, build_id, build_time, start_time, end_time, status)
-            time.sleep(5)
-            start_time = end_time
-
-        if status == "COMPLETED":
-            print("Deployment successful")
-            print(f"URL: {url}")
-            sys.exit(0)
-        else:
-            time.sleep(5)
-            check_service_status(token, build_id, build_time, start_time, datetime.now(pytz.timezone('UTC')), status)
-            print("Deployment failed")
-            sys.exit(1)
 
-    except Exception as e:
-        print("Error uploading package")
-        print(e)
+def upload_package(tar, token, env):
+    print("Uploading package to deployment server")
+    tar.seek(0)
+    files = {'file': tar}
+    start_time = datetime.now(pytz.timezone('UTC'))
+    build_time = start_time
+    with requests.post(
+        url = WRITER_DEPLOY_URL, 
+        headers = {
+            "Authorization": f"Bearer {token}",
+        },
+        files=files,
+        data={"envs": json.dumps(dictFromEnv(env))}
+    ) as resp:
+        on_error_print_and_raise(resp)
+        data = resp.json()
+        build_id = data["buildId"]
+
+    print("Package uploaded. Building...")
+    status = "WAITING"
+    url = ""
+    while status not in ["COMPLETED", "FAILED"] and datetime.now(pytz.timezone('UTC')) < build_time + timedelta(minutes=5):
+        end_time = datetime.now(pytz.timezone('UTC'))
+        status, url = check_service_status(token, build_id, build_time, start_time, end_time, status)
+        time.sleep(5)
+        start_time = end_time
+
+    if status == "COMPLETED":
+        print("Deployment successful")
+        print(f"URL: {url}")
+        sys.exit(0)
+    else:
+        time.sleep(5)
+        check_service_status(token, build_id, build_time, start_time, datetime.now(pytz.timezone('UTC')), status)
+        print("Deployment failed")
         sys.exit(1)
-    finally:
-        tar.close()
 
+def on_error_print_and_raise(resp):
+    try:
+        resp.raise_for_status()
+    except Exception as e:
+        if os.getenv('DEBUG') == 'true':
```

Comment:
I think a user might need to see the debug mode of the cli. What's your opinion on this topic ? I would suggest adding a `--debug` or `--verbose` flag which configures the `DEBUG` environment variable.